portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -25,28 +25,51 @@
 /** Servlet that returns comments. */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
-  public static final Gson GSON = new Gson();
+  private static final Gson GSON = new Gson();
+  private ArrayList<ArrayList<String>> messages = new ArrayList<>();
```

Comment:
This is not threadsaafe, while doPost happens concurrently.

Reply:
I'm not sure how I could solve this issue.
If I make messages a local variable in doPost(), is there a way for doGet() to access the variable?

Reply:
java provides multiple tricks for guarding threadsafety.
Maybe using synchronize or lock. You could search for examples.